src/loaders/zarr/pixel-source.ts

```diff
@@ -38,14 +38,19 @@ interface ZarrTileSelection {
 class ZarrPixelSource<S extends string[]> implements PixelSource<S> {
   private _data: ZarrArray;
   private _indexer: ZarrIndexer<S>;
+  private _readChunks: boolean;
 
-  constructor(data: ZarrArray, public labels: Labels<S>) {
+  constructor(
+    data: ZarrArray,
+    public labels: Labels<S>,
+    public tileSize: number
+  ) {
     this._indexer = getIndexer(labels);
     this._data = data;
-  }
 
-  get tileSize() {
-    return this._data.chunks[this._xIndex];
+    const xChunkSize = data.chunks[this._xIndex];
+    const yChunkSize = data.chunks[this._xIndex - 1];
+    this._readChunks = tileSize === xChunkSize && tileSize === yChunkSize;
```

Comment:
If there is a mismatch, we can't use the `getRawChunk` API from zarr.js. Need `slice` + `getRaw`.